src/MLJInterface.jl

```diff
@@ -406,7 +406,383 @@ MLJModelInterface.metadata_model(
+  features should be interpreted as categorical.
+- `data_random_seed::Int = 1`: Random seed used when constructing histogram bins.
+- `is_sparse::Bool = true`: Enable/disable sparse optimization.
+- `is_unbalance::Bool = false`: Set to true if training data is unbalanced.
+- `boost_from_average::Bool = true`: Adjusts the initial score to the mean of
+  labels for faster convergence.
+- `use_missing::Bool = true`: Whether or not to handle missing values.
+- `feature_pre_filter::Bool = true`: Whether or not to ignore unsplittable
+  features.
+- `alpha::Float64 = 0.9`: Parameter used for huber and quantile regression.
+- `metric::Vector{String} = ["l2"]`: Metric(s) to be used when evaluating on
+  evaluation set. For detailed information, see [the official
+  documentation](https://lightgbm.readthedocs.io/en/latest/Parameters.html#metric-parameters)
+- `metric_freq::Int = 1`: The frequency to run metric evaluation at.
+- `is_training_metric::Bool = false`: Set to `true` to output metric result on
+  training dataset.
+- `ndcg_at::Vector{Int} = Vector{Int}([1, 2, 3, 4, 5])`: Evaluation positions for
+  ndcg and map metrics.
+- `num_machines::Int = 1`: Number of machines to use when doing distributed
+  learning.
+- `num_threads::Int  = 0`: Number of threads to use.
+- `local_listen_port::Int = 12400`: TCP listen port.
+- `time_out::Int = 120`: Socket timeout.
+- `machine_list_file::String = ""`: Path of files that lists the machines used for
+  distributed learning.
+- `save_binary::Bool = false`: Whether or not to save the dataset to a binary file
+- `device_type::String = "cpu"`: The type of device being used. One of `cpu` or
+  `gpu`
+- `force_col_wise::Bool = false`: Force column wise histogram building. Only
+  applicable on cpu.
+- `force_row_wise::Bool = false`: Force row wise histogram building. Only
+  applicable on cpu.
+- `truncate_booster::Bool = true`: Whether or not to truncate the booster.
+
+# Fitted parameters
+
+The fields of `fitted_params(mach)` are:
+
+- `fitresult`: Fitted model information, contains a `LGBMRegression` object, an
+  empty vector, and the regressor with all its parameters
+
+# Report
+
+The fields of `report(mach)` are:
+
+- `training_metrics`: A dictionary containing all training metrics.
+- `importance`: A `namedtuple` containing:
+    - `gain`: The total gain of each split used by the model
+    - `split`: The number of times each feature is used by the model.
+
+# Examples
+
+```julia
+
+using DataFrames
+using MLJ
+
+# load the model (make sure to Pkg.add LightGBM to the environment)
+LGBMRegressor = @load LGBMRegressor
+
+X, y = @load_boston # a table and a vector 
+X = DataFrame(X)
+train, test = partition(collect(eachindex(y)), 0.70, shuffle=true)
+
+first(X, 3)
+lgb = LGBMRegressor() #initialised a model with default params
+mach = machine(lgb, X[train, :], y[train]) |> fit!
+
+predict(mach, X[test, :]) ```
+
+"""
+LGBMRegressor
```

Comment:
@ablaom is this LGBMRegressor required here (and the same for LGBMClassifier), or can these docs be placed above the actual LGBMRegressor and classifier structs?

Reply:
The reason they generally go there is explained [here](https://juliaai.github.io/MLJModelInterface.jl/dev/document_strings/#MLJModelInterface.doc_header). So it depends where your metadata declarations are.

I suggest you leave it where it is unless you have good reason not to.